### PR TITLE
feat: create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by qingzhuozhen**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'csharp', 'python' ]
```
Comment (haoliu-amp) [inline .github/workflows/codeql-analysis.yml:33]: Is C and objC options too?

Comment (dantetam) [reply .github/workflows/codeql-analysis.yml:33]: Yeah, I'm wondering how this would work with the complicated Unity Objective C++ mixed files (.m)

Comment (qingzhuozhen) [reply .github/workflows/codeql-analysis.yml:33]: C I believe is in option, objC however may be not according to https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/ . However, the code ql for this failed now.

**Comment by qingzhuozhen** [comment]
Close this as code ql not support objective c

**Pull request closed without merging**